apollo-router/src/axum_factory/listeners.rs

```diff
@@ -226,13 +229,20 @@ pub(super) fn serve_router_on_listen_addr(
     let (shutdown_sender, shutdown_receiver) = oneshot::channel::<()>();
 
     let meter = meter_provider().meter("apollo/router");
+    let total_session_count_instrument_address = address.to_string();
+
+    // This instrument is WRONG.
```

Comment:
I'm not sure to understand this comment

Reply:
It's that the connection addresses are not stored against the count which is a static. 

The address attribute is reported on the gauge, so at the very least listening on multiple addresses will always give the total across all addresses, but also after hot reload if the address changes then the metric is just reporting the connections against the new address when they could be against an old address.